core/src/main/java/org/kiwix/kiwixmobile/core/settings/SettingsPresenter.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.settings
+
+import android.util.Log
+import io.reactivex.CompletableObserver
+import io.reactivex.disposables.Disposable
+import org.kiwix.kiwixmobile.core.base.BasePresenter
+import org.kiwix.kiwixmobile.core.data.DataSource
+import org.kiwix.kiwixmobile.core.settings.SettingsContract.Presenter
+import org.kiwix.kiwixmobile.core.settings.SettingsContract.View
+import javax.inject.Inject
+
+internal class SettingsPresenter @Inject constructor(private val dataSource: DataSource) :
+  BasePresenter<View?>(), Presenter {
+  override fun clearHistory() {
+    dataSource.clearHistory()
+      .subscribe(object : CompletableObserver {
```

Comment:
Do me a favour and get rid of this CompletableObserver. There is a subscribe function that takes 2 lambdas, one that gets invoked on completing of the completable and one that gets invoked if there is an error